File: lab03/src/main/java/com/example/lab03/ui/quiz/MyCustomDialog.java
package com.example.lab03.ui.quiz;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.example.lab03.R;

public class MyCustomDialog extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        return new AlertDialog.Builder(requireContext())
                .setMessage(getString(R.string.custom_dialog_message))
                .setPositiveButton(getString(R.string.custom_dialog_button), (dialog, which) -> {} )
                .create();
    }

    public static String TAG = "MyCustomDialog";
}

File: lab04/src/main/java/com/example/lab04/ui/quiz/MyCustomDialog.kt
package com.example.lab04.ui.quiz

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import com.example.lab04.R;

class MyCustomDialog : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog =
        AlertDialog.Builder(requireContext())
            .setMessage(getString(R.string.custom_dialog_message))
            .setPositiveButton(getString(R.string.custom_dialog_button)) { _,_ -> }
            .create()

    companion object {
        const val TAG = "MyCustomDialog"
    }
}